crates/trippy-core/Cargo.toml

```diff
@@ -0,0 +1,61 @@
+[package]
+name = "trippy-core"
+description = "A network tracing library"
+version.workspace = true
+authors.workspace = true
+documentation.workspace = true
+homepage.workspace = true
+repository.workspace = true
+readme.workspace = true
+license.workspace = true
+edition.workspace = true
+rust-version.workspace = true
+keywords = ["traceroute", "ping", "icmp"]
+categories = ["network-programming"]
+
+[dependencies]
+derive_more.workspace = true
+thiserror.workspace = true
+tracing.workspace = true
+itertools.workspace = true
+arrayvec = { version = "0.7.4", default-features = false }
+socket2 = { version = "0.5.7", features = [ "all" ] }
```

Comment:
Bug fix: added feature `all` which is required.  It used to work by accident (another dependancy added that feature) but failed if you build `trippy-core` only in the virtual workspace.